webapp/src/app.tsx

```diff
@@ -2248,24 +2267,19 @@ function handleHash(hash: { cmd: string; arg: string }, loading: boolean): boole
             });
             window.location.hash = "";
             return true;
-        case "gettingstarted":
-            pxt.tickEvent("hash.gettingstarted")
-            editor.newProject();
-            window.location.hash = "";
-            return true;
-        case "tutorial":
+        case "tutorial": // shortcut to a tutorial. eg: #tutorial:tutorials/getting-started
             pxt.tickEvent("hash.tutorial")
             editor.startTutorial(hash.arg);
             window.location.hash = "";
```

Comment:
I think in all these places we should be using `Location.replace`, so it won't store the entry in browser history.

(except when the user actually clicks on something to go from home to editor or back - then `Location.assign` so history entry is stored)

Reply:
I went with replaceState instead of replace, and kept using location.hash = "" to make sure we keep an entry stored. Any reason to use replace and assign?